fenix/app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt

```diff
@@ -92,12 +95,16 @@ class HistoryFragment : LibraryPageFragment<History>(), UserInteractionHandler,
         val view = binding.root
         historyStore = StoreProvider.get(this) {
             HistoryFragmentStore(
-                HistoryFragmentState(
-                    items = listOf(),
-                    mode = HistoryFragmentState.Mode.Normal,
-                    pendingDeletionItems = emptySet(),
-                    isEmpty = false,
-                    isDeletingItems = false,
+                initialState = HistoryFragmentState.initial,
+                middleware = listOf(
+                    HistoryNavigationMiddleware(
+                        navController = findNavController(),
+                        openToBrowser = ::openItem,
+                    ),
+                    HistoryTelemetryMiddleware(
+                        isInPrivateMode = (activity as HomeActivity)
+                            .browsingModeManager.mode == BrowsingMode.Private,
```

Comment:
if you're leery about using the activity still, the mode is also available in the `AppStore`